packages/parameters/tests/unit/SSMProvider.test.ts

```diff
@@ -1310,4 +1312,92 @@ describe('Class: SSMProvider', () => {
       });
     });
   });
+
+  describe('Method: _set', () => {
+    test('sets a parameter successfully', async () => {
+      const provider: SSMProvider = new SSMProvider();
+      const client = mockClient(SSMClient)
+        .on(PutParameterCommand)
+        .resolves({ Version: 1 });
+      const parameterName: string = '/my-parameter';
+      const options: SSMSetOptions = { value: 'my-value' };
+
+      const version = await provider.set(parameterName, options);
+
+      expect(version).toBe(1);
+      expect(client).toReceiveCommandWith(PutParameterCommand, {
+        Name: parameterName,
+        Value: options.value,
+      });
+    });
+
+    test('returns undefined if version is undefined', async () => {
```

Comment:
Can the version ever be undefined? I thought that the successful creation of the parameter would always return a version number/id

Reply:
The `Version` in `PutParameterResult` can be undefined, so I thought i better test it.
Source: https://github.com/aws/aws-sdk-js-v3/blob/39532e4f960bf964f059f0f1d633829675e7f92b/clients/client-ssm/src/models/models_1.ts#L9665

Reply:
I see - the type in the SDK does put it as potentially undefined but the [API seems to suggest it's always present](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_PutParameter.html#API_PutParameter_ResponseSyntax) whenever the request is successful.

I'm ambivalent on this, I think the API is the source of truth but the check is not expensive so I leave it up to you.

Reply:
I also think it would be a better DX if I can assume that the setParameter always returns _something_
How would you deal with this potential undefined value code-wise? Biome prevents non-null assertions and I'm not sure if setting a default value is correct here ( I doubt it )

Reply:
For now I added a biome-ignore

Reply:
I think that's sensible, as long as we leave a one-sentence explanation of why for future reference.